### PR TITLE
feat(vertexai): update function calling and sample

### PR DESCRIPTION
**Pull request by cynthiajoan**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart [hunk 1]
```diff
@@ -21,7 +21,13 @@ import 'vertex_schema.dart';
 /// knowledge and scope of the model.
 final class Tool {
   /// Constructor
-  Tool({this.functionDeclarations});
+  Tool._({this.functionDeclarations});
+
+  /// Returns a [Tool] instance with list of [FunctionDeclaration].
+  static Tool functionDeclarationsTool(
```
Comment (natebosch) [inline packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart:27]: I don't think it is useful to repeate `Tool` in the name.

how about `static Tool functionDeclarations(`

**Review by natebosch — APPROVED** [review]

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart (hunk 1 above)
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart:27]: It's same with the attributes. Would you suggest to make the attribute functionDeclarations private?

File: packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart [hunk 2]
```diff
@@ -91,52 +91,77 @@ class _ChatWidgetState extends State<ChatWidget> {
...
+          Tool.functionDeclarationsTool([getWeatherTool]),
         ],
       );
       _chat = _model.startChat();
     });
   }
 
-  Future<Map<String, Object?>> findExchangeRate(
+  // This is a hypothetical API to return a fake weather data collection for certain location
+  Future<Map<String, Object?>> getWeather(
```
[the remark this answers is not in the document]
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart]: Think over this, and would like to keep getWeather to keep consistency with the devsite documentation.

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart (hunk 1 above)
Comment (natebosch) [reply packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_function_calling.dart:27]: Hmm, good point. I guess I might skip the static method altogether in that case. It's not really saving any characters compared to the constructor and named argument.

**Pull request merged**